sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientJavaDocCodeSnippets.java

```diff
@@ -24,4 +43,892 @@ public TextAnalyticsAsyncClient createTextAnalyticsAsyncClient() {
         // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.instantiation
         return textAnalyticsAsyncClient;
     }
+
+    // Languages
+    /**
+     * Code snippet for detecting Language
+     */
+    public void detectLanguageCodeSnippets() {
+        TextAnalyticsAsyncClient textAnalyticsAsyncClient = createTextAnalyticsAsyncClient();
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectLanguage#string
+        String inputText = "Bonjour tout le monde";
+        textAnalyticsAsyncClient.detectLanguage(inputText).subscribe(detectLanguageResult -> {
+            for (DetectedLanguage detectedLanguage : detectLanguageResult.getDetectedLanguages()) {
+                System.out.printf("Detected languages name: %s, ISO 6391 Name: %s, Score: %s.%n",
+                    detectedLanguage.getName(),
+                    detectedLanguage.getIso6391Name(),
+                    detectedLanguage.getScore());
+            }
+        });
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectLanguage#string
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectLanguageWithResponse#string-string
+        String input = "This text is in English";
+        String countryHint = "US";
+        textAnalyticsAsyncClient.detectLanguageWithResponse(input, countryHint).subscribe(detectLanguageResult -> {
+            for (DetectedLanguage detectedLanguage : detectLanguageResult.getValue().getDetectedLanguages()) {
+                System.out.printf("Detected languages name: %s, ISO 6391 Name: %s, Score: %s.%n",
+                    detectedLanguage.getName(),
+                    detectedLanguage.getIso6391Name(),
+                    detectedLanguage.getScore());
+            }
+        });
+        // END: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectLanguageWithResponse#string-string
+
+        // BEGIN: com.azure.ai.textanalytics.TextAnalyticsAsyncClient.detectLanguages#List
+        final List<String> inputs = Arrays.asList(
+            "This is written in English", "Este es un document escrito en Español.");
+        textAnalyticsAsyncClient.detectLanguages(inputs).subscribe(detectedBatchResult -> {
+            System.out.printf("Model version: %s%n", detectedBatchResult.getModelVersion());
+
+            // Batch statistics
+            final TextDocumentBatchStatistics batchStatistics = detectedBatchResult.getStatistics();
+            System.out.printf("Batch statistics, document count: %s, erroneous document count: %s,"
+                    + " transaction count: %s, valid document count: %s.%n",
+                batchStatistics.getTransactionCount(),
+                batchStatistics.getTransactionCount(),
+                batchStatistics.getValidDocumentCount());
+
+            for (DetectLanguageResult detectLanguageResult : detectedBatchResult) {
+                for (DetectedLanguage detectedLanguage : detectLanguageResult.getDetectedLanguages()) {
```

Comment:
System.out.printf("Document ID: %s%n", detectLanguageResult.getId());

Reply:
we need erroneous documents as well. Otherwise, if detectLanguageResult is error, `detectLanguageResult.getDetectedLanguages()` will have NPE
```
            // Erroneous document
            if (detectLanguageResult.isError()) {
                System.out.printf("Cannot detect language. Error: %s%n", detectLanguageResult.getError().getMessage());
                continue;
            }
```

Reply:
These are code snippets that would be inserted in the method java docs. I think we should keep them shorter and to the point, that shows how to use the method. We don't need to handle all the fail through scenarios here, samples should be a good place to do that. What do you think?

Reply:
Ya. Let's keep this codesnippet shorter.